### DISCUSSION TITLE
Enable/Disable Rows in Ag Grid

### DISCUSSION THREAD

**Discussion by AshCusack** [zauberzeug/nicegui#1577] [Q&A] [answered]
### Question

I've got a ag grid that I am programmatically adding rows to. Is there away to disable rows based on criteria of the incoming data.

Tried API calls and various methods with no success.
I am adding the data using a dictionary variable and appending to the rowData list, perhaps this isn't the correct way?

Below is an image, I want the T1-CIN entry to be disabled and not checkable.

![image](https://github.com/zauberzeug/nicegui/assets/22345595/25a46a39-623d-467c-846f-bf023a4d9e95)

Thanks

**Comment by falkoschindler** [comment]
Do you know how it could be done with JavaScript? Once we find out how to do it with the plain AG Grid framework, there's certainly a way to do it with NiceGUI.

**Reply by AshCusack** [reply]
It would seem rather than disabled a specific row, you can create parameters whereby a row will be disabled if it matches the criteria. This is done in the grid options by the following
` isRowSelectable: (params) => {
    return !!params.data && params.data.year === 2012;
  },`
(Taken from: https://www.ag-grid.com/javascript-data-grid/row-selection/)

Not sure how I would convert this into nicegui.

**Reply by falkoschindler** [reply, marked as answer]
Functions like this can be passed to `ui.aggrid` as a string. A leading colon ":" indicates that this attribute needs to be evaluated on the client:
```py
ui.aggrid({
    ...
    ':isRowSelectable': 'params => params.data.year === 2012',
})
```

**Reply by AshCusack** [reply]
Thank you so much.
